### DISCUSSION TITLE
I can't open App Installer from Microsoft Store Apps page.

### DISCUSSION THREAD

**Discussion by doggy8088** [microsoft/winget-cli#2465] [General]
I'm at [Use the winget tool to install and manage applications](https://docs.microsoft.com/en-us/windows/package-manager/winget/?WT.mc_id=DT-MVP-4015686) page. When I clicked [get App Installer from the Microsoft Store](https://www.microsoft.com/p/app-installer/9nblggh4nns1#activetab=pivot:overviewtab) link. I always get `Not Available for bots to index` message.

![image](https://user-images.githubusercontent.com/88981/185791936-fc96ceb8-f083-4cd9-af07-f0249a23a82e.png)

I checked the URL. It's been redirected from <https://www.microsoft.com/p/app-installer/9nblggh4nns1#activetab=pivot:overviewtab> to <https://apps.microsoft.com/store/detail/9NBLGGH4NNS1?hl=zh-tw&gl=TW#activetab=pivot:overviewtab>.

I actually can remove the `?hl=zh-tw&gl=TW` from URL to get it working. But I don't know why I have to that in the URL.

![image](https://user-images.githubusercontent.com/88981/185792077-61761429-d0e1-4903-b8b8-a2233c648ece.png)

I'm not sure where to report this issue. Hope the Microsoft guys can report this issue to them.

**Comment by gyp** [comment]
It's not specific to AppInstaller -- we've been seeing the same issue with our own app for the past few days. It seems to be a server-side rate-limiting/caching issue at Microsoft as changing _anything_ in the URL fixes the problem temporarily -- only to get back to the same issue a few hours later. We also found that VPN-ing to some locations (mostly to the US) helped as well. As a temporary solution we ended up adding the timestamp to the link, but that does not seem like a scalable solution to be honest.

We've reached out to our MS contact about this but we haven't heard back yet.

**Reply by V3NG3URZ** [reply]
Hi do you have any update about this ? the download from microsoft store is banned for normal users for my company so we download package with a powershell script for the users when it was validate by security.

The issue here is that we use index of processed filename to download each package 

but with " Not available for bots to index" we have an issue with NULL index 

so if you have a solution from MS tell us :)

**Reply by gyp** [reply]
Not yet -- they're still looking into it. So far just adding a random additional query parameter to the URL worked out just fine and got rid of these errors.

**Reply by naman-0804** [reply]
I cant even find whatsapp beta on store and the link to it gives this as an error

**Reply by V3NG3URZ** [reply]
> 

thanks for the answer. @gyp DO you have exemple of what type of query do you ad ? i use this kind of links : 
"https://www.microsoft.com/fr-fr/p/qr-code-for-windows-10/9nblggh5m02g"
"https://www.microsoft.com/p/heif-image-extensions/9pmmsr1cgpwg" 

thank you

**Reply by gyp** [reply]
This is what we link to:

https://apps.microsoft.com/store/detail/shapr3d/9N4K9QFV4XFC?gclid=f-3579a28842a7bcac7ba630d698829e9b

Where the "f-...."  is a pseudorandom string (basically the md5 hash of the timestamp, but practically anything would do). It's an ugly crude hack, but seems to be working and better than a good percentage of our users going to a dead end.

**Reply by V3NG3URZ** [reply]
i still got the issue when i add it to my links

**Reply by gyp** [reply]
I'm sorry to hear that. It worked for us.

As it seems to be something that happens due to a setting on the MS side, I've got zero insight into what's going on so I can't offer you any more tips. I'll make sure to share  any info when I hear back from them.

**Reply by V3NG3URZ** [reply]
no problem thank you for your help
